python/GafferArnoldTest/IECoreArnoldPreviewTest/RendererTest.py

```diff
@@ -226,6 +226,69 @@ def testShaderNames( self ) :
 			self.assertEqual( len( shaders ), 4 )
 			self.assertEqual( len( set( [ arnold.AiNodeGetName( s ) for s in shaders ] ) ), 4 )
 
+	def testShaderNodeConnectionType( self ) :
+
+		r = GafferScene.Private.IECoreScenePreview.Renderer.create(
+			"IECoreArnold::Renderer",
+			GafferScene.Private.IECoreScenePreview.Renderer.RenderType.SceneDescription,
+			self.temporaryDirectory() + "/test.ass"
+		)
+
+		scalarColorShader = IECore.ObjectVector( [
+			IECore.Shader( "image", parameters = { "__handle" : "scalarColorSource" } ),
+			IECore.Shader( "lambert", parameters = { "Kd_color" : "link:scalarColorSource", "__handle" : "scalarColorTarget" } ),
+		] )
+
+		scalarNodeShader = IECore.ObjectVector( [
+			IECore.Shader( "gobo", parameters = { "__handle" : "scalarNodeSource" } ),
+			IECore.Shader( "persp_camera", parameters = { "filtermap" : "link:scalarNodeSource", "__handle" : "scalarNodeTarget" } ),
+		] )
+
+		vectorColorShader = IECore.ObjectVector( [
+			IECore.Shader( "image", parameters = { "__handle" : "vectorColorSource" } ),
+			IECore.Shader( "htoa__ramp_rgb", parameters = { "color" : IECore.StringVectorData(["link:vectorColorSource"]), "__handle" : "vectorColorTarget" } ),
```

Comment:
Not that I could find by grepping kick -info.  Should I just remove the test?  Doesn't seem worth building a custom C++ shader node.  Maybe it could be done with an OSL shader?